model-optimizer/extensions/front/onnx/loop_ext.py

```diff
@@ -105,8 +176,11 @@ def extract(cls, loop_node):
             body_results.append(Node(body_graph, add_opoutput(body_graph, node_name, output_port, False)))
 
         # add 'internal_layer_id' attribute which is a must have attribute for the loop body node
-        for idx, body_node in enumerate(body_graph.get_op_nodes()):
-            body_node['internal_layer_id'] = idx
+        cur_graph = body_graph
+        while 'parent_node' in cur_graph.graph:
+            for idx, body_node in enumerate(cur_graph.get_op_nodes()):
+                body_node['internal_layer_id'] = idx
+            cur_graph = cur_graph.graph['parent_node'].graph
```

Comment:
Am I correct that we have to re-enumerate layer ids due to possible newly appeared transit nodes?
If yes, maybe we should define `internal_layer_id` in-fly while creating it so the previous version of `for` loop will be sufficient

Reply:
We re-enumerate layers for all loops because transit parameter can be added. Changed this logic to avoid re-enumerating all bodies

Reply:
Great!